pkg/shares/non_interactive_defaults.go

```diff
@@ -83,10 +83,31 @@ func roundUpBy(cursor, v int) int {
 	}
 }
 
-// MinSquareSize returns the minimum square size that can contain shareCount
-// number of shares.
-func MinSquareSize(shareCount int) int {
-	return RoundUpPowerOfTwo(int(math.Ceil(math.Sqrt(float64(shareCount)))))
+// SubTreeWidth determines how many shares should be included when creating
+// subtree. This is used for determining which subtree roots are used in the
+// share commitment for PFBs, and therefore also the padding between blobs. The
+// reasoning behind this algorithm is discussed in depth in ADR013.
+func SubTreeWidth(shareCount int) int {
+	// per ADR013, we use a predetermined threshold to determine width of sub
+	// trees used to create share commitments
+	s := (shareCount / appconsts.SubtreeRootHeightThreshold)
+
+	// round up if the width is not an exact multiple of the threshold
+	if shareCount%appconsts.SubtreeRootHeightThreshold != 0 {
+		s++
+	}
+
+	// use a power of two equal to or larger than the multiple of the subtree
+	// root threshold
+	s = RoundUpPowerOfTwo(s)
+
+	// using a value below the min square size is wasteful, see ADR013 for more
+	// details
+	if s < appconsts.DefaultMinSquareSize {
+		return appconsts.DefaultMinSquareSize
+	}
+
+	return s
```

Comment:
this is the only consensus logic we have to change!! the rest is just fixing tests